components/match2/commons/template_match.lua

```diff
@@ -114,6 +114,8 @@ function TemplateMatch._recursiveSetBracketIndex(matches, id, headerchild, appli
 		headerchild = true
 	end
 	match.bracketdata.bracketindex = applied
+	--id can not be nil
+	---@diagnostic disable-next-line: need-check-nil
```

Comment:
Code should be updated
cast -nil after logic sciope
remove the id nil check and return (and add annotation that requires string input